src/main/java/com/alex/eshop/eshop/Repository/ItemRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.alex.eshop.eshop.Repository;
+
+import com.alex.eshop.eshop.Entity.Item;
+
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface ItemRepository extends JpaRepository<Item, Integer> {
+    @Query("select i from Item i order by i.id desc")
+    List<Item> getLastFiveItems();
+    Optional<Item> findById(Long id);
```

Comment:
Function `Optional<Item> findById(Long id);` is redundant, because `interface CrudRepository` already have this function. You cannot use it because you defined wrong ID Type in `JpaRepository<Item, Integer>`. It's not Integer.